bin/wiki_entity_linking/entity_linker_evaluation.py

```diff
@@ -0,0 +1,247 @@
+
+    logger.info(baseline_accuracies.report_accuracy("random"))
+    logger.info(baseline_accuracies.report_accuracy("prior"))
+    logger.info(baseline_accuracies.report_accuracy("oracle"))
+
+    # using only context
+    el_pipe.cfg["incl_context"] = True
+    el_pipe.cfg["incl_prior"] = False
+    dev_precision_context, dev_recall_context, dev_acc_cont_d = measure_acc(dev_data, el_pipe)
+    print("dev precision context avg:", round(dev_precision_context, 3))
+    print("dev recall context avg:", round(dev_recall_context, 3))
+
+    # measuring combined accuracy (prior + context)
+    el_pipe.cfg["incl_context"] = True
+    el_pipe.cfg["incl_prior"] = True
+    dev_precision_combo, dev_recall_combo, dev_acc_combo_d = measure_acc(dev_data, el_pipe)
+    print("dev precision combo avg:", round(dev_precision_combo, 3))
+    print("dev recall combo avg:", round(dev_recall_combo, 3))
+
+
+def measure_acc(data, el_pipe=None, error_analysis=False):
+    # If the docs in the data require further processing with an entity linker, set el_pipe
+    correct_by_label = defaultdict(int)
+    incorrect_by_label = defaultdict(int)
+
+    docs = []
+    golds = []
+    for d, g in tqdm(data, leave=False):
+        if len(d) > 0:
+            docs.append(el_pipe(d))
+            golds.append(g)
+
+    false_positives = 0
+    true_positives = 0
+    false_negatives = 0
+    for doc, gold in zip(docs, golds):
+        try:
+            correct_entries_per_article = dict()
+            for entity, kb_dict in gold.links.items():
+                start, end = entity
+                # only evaluating on positive examples
+                for gold_kb, value in kb_dict.items():
+                    if value:
+                        offset = _offset(start, end)
+                        correct_entries_per_article[offset] = gold_kb
+
+            for ent in doc.ents:
+                ent_label = ent.label_
+                pred_entity = ent.kb_id_
+                start = ent.start_char
+                end = ent.end_char
+                offset = _offset(start, end)
+                gold_entity = correct_entries_per_article.get(offset, None)
+                # the gold annotations are not complete so we can't evaluate missing annotations as 'wrong'
+                if gold_entity is not None:
+                    if gold_entity == pred_entity:
+                        correct_by_label[ent_label] += 1
+                        true_positives += 1
+                    elif pred_entity == "NIL":
+                        false_negatives += 1
+                    else:
+                        incorrect_by_label[ent_label] += 1
+                        false_positives += 1
+                        false_negatives += 1
+
+        except Exception as e:
+            print("Error assessing accuracy", e)
+
+    _, acc_by_label = calculate_acc(correct_by_label, incorrect_by_label)
+    return (
```

Comment:
I think your switch to `precision` and `recall` makes sense. `accuracy` is anyway confusing because you can't really define true negatives with this data. So in the definition we have here, accuracy equals precision, right? Should we go ahead then and remove the references to accuracy/`acc` and call all of it `precision`, also the `_by_label` dictionaries?

Reply:
> So in the definition we have here, accuracy equals precision, right?

That's right, precision here is equivalent to how accuracy was defined before.

> Should we go ahead then and remove the references to accuracy/acc and call all of it precision, also the _by_label dictionaries?

That sounds good to me!

Reply:
so the _by_label dictionaries are completely removed now? I thought it was kind of informative to break up the performance by NER label, because it can vary quite a bit...

Reply:
I think I might have misunderstood the "also the _by_label dictionaries" in your earlier comment. I understood it to mean we remove all references to the `_by_label` dictionaries. I can add them back but with precision and recall 👍

Reply:
Oh, sorry that that was unclear. I was indeed just proposing to change the terminology, following your initial change from acc to p, which made sense...